File: frontend/between_pages/lib/models/user/user_response_dto.dart
class UserResponseDTO {
  final int idUser;
  final String name;
  final String email;
  final String role;

  UserResponseDTO({
    required this.idUser,
    required this.name,
    required this.email,
    required this.role,
  });

  factory UserResponseDTO.fromJson(Map<String, dynamic> json) {
    return UserResponseDTO(
      idUser: int.tryParse(json['id']?.toString() ?? '0') ?? 0,
      name: json['name'] as String? ?? 'Usuario',
      email: json['email'] as String? ?? '',
      role: json['role'] as String? ?? 'USER',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': idUser,
      'name': name,
      'email': email,
      'role': role,
    };
  }
}

File: backend/src/main/java/com/calonuria/backend/dto/user/UserResponseDTO.java
package com.calonuria.backend.dto.user;

import lombok.Data;
import lombok.AllArgsConstructor;

/**
 * DTO para la respuesta con información de usuario.
 */
@Data
@AllArgsConstructor
public class UserResponseDTO {

    /**
     * ID del usuario.
     */
    private Long id;

    /**
     * Nombre del usuario.
     */
    private String name;

    /**
     * Correo electrónico del usuario.
     */
    private String email;

    /**
     * Rol del usuario (USER o ADMIN).
     */
    private String role;
}